modules/base/rendering/renderablemodel.cpp

```diff
@@ -549,31 +653,68 @@ void RenderableModel::render(const RenderData& data, RendererTasks&) {
         case ColorAddingBlending:
             glBlendFunc(GL_SRC_COLOR, GL_DST_COLOR);
             break;
-    };
+        };
 
-    if (_disableDepthTest) {
-        glDisable(GL_DEPTH_TEST);
-    }
+        if (_disableDepthTest) {
+            glDisable(GL_DEPTH_TEST);
+        }
 
-    _geometry->render(*_program);
-    if (_disableFaceCulling) {
-        glEnable(GL_CULL_FACE);
-    }
+        _geometry->render(*_program);
+        if (_disableFaceCulling) {
+            glEnable(GL_CULL_FACE);
+        }
 
-    global::renderEngine->openglStateCache().resetBlendState();
+        global::renderEngine->openglStateCache().resetBlendState();
 
-    if (_disableDepthTest) {
-        glEnable(GL_DEPTH_TEST);
-    }
+        if (_disableDepthTest) {
+            glEnable(GL_DEPTH_TEST);
+        }
 
-    _program->deactivate();
+        _program->deactivate();
+    }
 }
 
-void RenderableModel::update(const UpdateData&) {
+void RenderableModel::update(const UpdateData& data) {
     if (_program->isDirty()) {
         _program->rebuildFromFile();
         ghoul::opengl::updateUniformLocations(*_program, _uniformCache, UniformNames);
     }
+
+    if (_geometry->hasAnimation() && _animationStart != "") {
+        double realtiveTime;
+        double now = data.time.j2000Seconds();
+        double startTime = data.time.convertTime(_animationStart);
+        double duration = _geometry->animationDuration();
+
+        switch (_animationMode) {
+            case AnimationMode::LoopFromStart:
+                realtiveTime = std::fmod(now - startTime, duration);
+                break;
+            case AnimationMode::LoopInfinitely:
+                realtiveTime = std::fmod(now - startTime, duration);
+                if (realtiveTime < 0) {
+                    realtiveTime += duration;
+                }
+                break;
+            case AnimationMode::BounceFromStart:
+                realtiveTime =
+                    duration - abs(fmod(now - startTime, 2 * duration) - duration);
+                break;
+            case AnimationMode::BounceInfinitely: {
+                double modulo = fmod(now - startTime, 2 * duration);
+                if (modulo < 0) {
+                    modulo += 2 * duration;
+                }
+                realtiveTime = duration - abs(modulo - duration);
+                break;
+            }
+            case AnimationMode::Once:
+            default:
```

Comment:
Maybe better to move the code actually into the `Once` case and throw a `ghoul::MissingCaseException` here in the `default` label